packages/cli/src/controllers/users.controller.ts

```diff
@@ -295,6 +297,15 @@ export class UsersController {
 
 		await this.userService.update(targetUser.id, { role: payload.newRoleName });
 
+		const adminDowngradedToMember =
+			req.user.role === 'global:owner' &&
+			targetUser.role === 'global:admin' &&
+			payload.newRoleName === 'global:member';
+
+		if (adminDowngradedToMember) {
```

Comment:
Since the update to the user and to the keys don't happen inside a transaction we can end up with the user being downgraded, but the api key update failing. 

I think we can either do it in reverse, first downgrade the keys and then the user, then we have to downgrade the keys within a transaction, then we only end up with users that are still admins, but their keys have been downgraded instead of the reverse, or we run all of it inside a transaction.